examples/experimental/dagster-airlift/dagster_airlift/in_airflow/base_proxy_operator.py

```diff
@@ -63,127 +98,225 @@ def _get_validated_session(self, context: Context) -> requests.Session:
             )
         return session
 
-    @abstractmethod
-    def get_dagster_url(self, context: Context) -> str:
-        """Returns the URL for the Dagster instance."""
-
-    def get_valid_graphql_response(self, response: Response, key: str) -> Any:
-        response_json = response.json()
-        if not response_json.get("data"):
-            raise Exception(f"Error in GraphQL request. No data key: {response_json}")
-
-        if key not in response_json["data"]:
-            raise Exception(f"Error in GraphQL request. No {key} key: {response_json}")
-
-        return response_json["data"][key]
-
-    def launch_runs_for_task(self, context: Context, dag_id: str, task_id: str) -> None:
-        """Launches runs for the given task in Dagster."""
-        session = self._get_validated_session(context)
+    def get_dag_run(self) -> DagRun:
+        context = get_current_context()
+        if "dag_run" not in context:
+            raise Exception("Dag run not found in context. Expected to find dag_run.")
+        return context["dag_run"]
+
+    def get_dag_id(self) -> str:
+        return self.get_dag_run().dag_id  # type: ignore # thinks it's a column due to django syntax
+
+    def get_dag_run_id(self) -> str:
+        return self.get_dag_run().run_id  # type: ignore # thinks it's a column due to django syntax
+
+    def get_task_id(self) -> str:
+        context = get_current_context()
+        if "task" not in context:
+            raise Exception("Task not found in context. Expected to find task.")
+        return context["task"].task_id
+
+    def get_dagster_run_tags(self) -> Mapping[str, str]:
+        dag_run = self.get_dag_run()
+        return {
+            DAG_ID_TAG_KEY: cast(str, dag_run.dag_id),
+            DAG_RUN_ID_TAG_KEY: cast(str, dag_run.run_id),
+            TASK_ID_TAG_KEY: self.get_task_id(),
+        }
 
-        dagster_url = self.get_dagster_url(context)
-        assets_to_trigger = {}  # key is (repo_location, repo_name, job_name), value is list of asset keys
-        # create graphql client
-        response = session.post(
+    @lru_cache(maxsize=1)
```

Comment:
I don't think this is safe because we don't know the process lifecycle across all airflow installations. Instead just ensure that it isn't called more than once per task run